archive/2025-09-23-trusted-publishing-for-npm-packages-npm-docs.md

```diff
@@ -0,0 +1,320 @@
+  secure CI/CD workflows without long-lived tokens. This standard, endorsed by Open
+  Source Security Foundation, improves security by using short-lived, cryptographically-signed
+  tokens tailored to specific workflows—reducing risks associated with token management.
+  Supported initially by GitHub Actions and GitLab CI/CD, this mechanism supports
+  streamlined package publishing and provenance generation, enhancing authenticity
+  verification. Best practices recommend restricting traditional token access for
+  greater security. Future enhancements may include support for self-hosted runners
+  and broader CI/CD provider integration.
+link: https://docs.npmjs.com/trusted-publishers
+tags:
+- OIDC
+- security best practices
+- trusted publishing
+- CI/CD
+- npm
+title: Trusted publishing for npm packages ◆ npm Docs
+---
+
+[Skip to search](https://docs.npmjs.com/trusted-publishers#search-box-input) [Skip to content](https://docs.npmjs.com/trusted-publishers#skip-to-content)
+
+### Site navigation
+
+# Trusted publishing for npm packages
+
+Table of contents
+
+Trusted publishing allows you to publish npm packages directly from your CI/CD workflows using [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) authentication, eliminating the need for long-lived npm tokens. This feature implements the [trusted publishers industry standard](https://repos.openssf.org/trusted-publishers-for-all-package-repositories) specified by the Open Source Security Foundation (OpenSSF), joining a growing ecosystem including [PyPI](https://docs.pypi.org/trusted-publishers/), [RubyGems](https://guides.rubygems.org/trusted-publishing/), and other major package registries in offering this security enhancement.
+
+**Note:** Trusted publishing requires [npm CLI](https://docs.npmjs.com/cli/v11) version 11.5.1 or later.
+
+## [How trusted publishing works](https://docs.npmjs.com/trusted-publishers\#how-trusted-publishing-works)
+
+Trusted publishing creates a trust relationship between npm and your CI/CD provider using OIDC. When you configure a trusted publisher for your package, npm will accept publishes from the specific workflow you've authorized, in addition to traditional authentication methods like npm tokens and manual publishes. The npm CLI automatically detects OIDC environments and uses them for authentication before falling back to traditional tokens.
+
+This approach eliminates the security risks associated with long-lived write tokens, which can be compromised, accidentally exposed in logs, or require manual rotation. Instead, each publish uses short-lived, cryptographically-signed tokens that are specific to your workflow and cannot be extracted or reused.
+
+## [Supported CI/CD providers](https://docs.npmjs.com/trusted-publishers\#supported-cicd-providers)
+
+Trusted publishing currently supports:
+
+- [GitHub Actions](https://github.com/features/actions) (GitHub-hosted runners)
```

Comment:
The benefits are obvious. But are we losing control? All these "managed identities" usually fail to provide the same level of logging and traceability we expect when we manage our own identities.